orttraining/orttraining/training_api/module.cc

```diff
@@ -48,62 +49,103 @@ Status Parameter::ResetGrad() {
   }
   return Status::OK();
 }
-Module::Module(const std::string& train_model_path_or_bytes,
-               std::unordered_map<std::string, std::shared_ptr<Parameter>>& parameters,
-               const std::optional<std::string>& eval_model_path_or_bytes) : named_parameters_(parameters) {
-  // create value copy as same params are passed to Optimizer constructor
-
-  auto so = onnxruntime::SessionOptions();
-  std::unique_ptr<Environment> env;
-  ORT_THROW_IF_ERROR(Environment::Create(nullptr, env));
-  train_sess_ = std::make_unique<onnxruntime::InferenceSession>(so, *env);
-  ORT_THROW_IF_ERROR(train_sess_->Load(train_model_path_or_bytes));
-  ORT_THROW_IF_ERROR(train_sess_->Initialize());
-  if (eval_model_path_or_bytes.has_value()) {
-    eval_sess_ = std::make_unique<onnxruntime::InferenceSession>(so, *env);
-    ORT_THROW_IF_ERROR(eval_sess_->Load(eval_model_path_or_bytes.value()));
-    ORT_THROW_IF_ERROR(eval_sess_->Initialize());
-    utils::GetGraphInputOutputNames(eval_sess_, eval_input_names_, eval_output_names_);
-  }
 
+Module::Module(std::unordered_map<std::string, std::shared_ptr<Parameter>>& named_parameters,
+               InferenceSession* train_session) {
+  train_sess_ = train_session;
```

Comment:
nullptr check. Also let's not pass around raw pointers, see comment in optimizer.cc

Reply:
this will change in my PR... module will go back to creating the inference session internally